lxd/network_acls.go

```diff
@@ -184,25 +228,34 @@ func networkACLsGet(d *Daemon, r *http.Request) response.Response {
 	resultString := []string{}
 	resultMap := []*api.NetworkACL{}
 	urlToNetworkACL := make(map[*api.URL]auth.EntitlementReporter)
-	for _, aclName := range aclNames {
-		if !userHasPermission(entity.NetworkACLURL(requestProjectName, aclName)) {
-			continue
-		}
-
-		if !recursion {
-			resultString = append(resultString, fmt.Sprintf("/%s/network-acls/%s", version.APIVersion, aclName))
-		} else {
-			netACL, err := acl.LoadByName(s, effectiveProjectName, aclName)
-			if err != nil {
+	for projectName, acls := range aclNames {
+		for _, aclName := range acls {
+			if !userHasPermission(entity.NetworkACLURL(projectName, aclName)) {
 				continue
 			}
 
-			netACLInfo := netACL.Info()
-			netACLInfo.UsedBy, _ = netACL.UsedBy() // Ignore errors in UsedBy, will return nil.
-			netACLInfo.UsedBy = project.FilterUsedBy(s.Authorizer, r, netACLInfo.UsedBy)
-
-			resultMap = append(resultMap, netACLInfo)
-			urlToNetworkACL[entity.NetworkACLURL(requestProjectName, aclName)] = netACLInfo
+			if !recursion {
+				resultString = append(resultString, fmt.Sprintf("/%s/network-acls/%s", version.APIVersion, aclName))
```

Comment:
just a nit/suggestion for a follow up PR, but here the URL could be generated using the URL builder to ensure aclName is path escaped.